### PR TITLE
Msg send fail

### PR DESCRIPTION
**Pull request by tungsheng**
# Summary

Add send failed msg.

### CODE REVIEW

File: src/Bubble.js [hunk 1]
```diff
@@ -139,6 +145,17 @@ export default class Bubble extends React.Component {
     }
     return null;
   }
+  renderStatus() {
+    if (this.props.currentMessage.status === MSG_STATUS.FAIL) {
+      const statusStyle = {
+        marginVertical: 2,
+        marginHorizontal: 5,
+        color: 'red',
```
Comment (howieyoung) [inline src/Bubble.js]: Let's give it a warmer color 'rgb(221, 38, 75)'
and a fontSize 12

File: src/Bubble.js [hunk 2]
```diff
@@ -139,6 +145,17 @@ export default class Bubble extends React.Component {
...
+      };
+      return <Text style={statusStyle}>send failed</Text>;
```
Comment (howieyoung) [inline src/Bubble.js]: I think it should be 'Failed to send'?

File: src/Bubble.js [hunk 3]
```diff
@@ -23,6 +23,12 @@ import {isSameUser, isSameDay} from './utils';
 
 const {width} = Dimensions.get('window'); // 2019.03.20 Tony Lee
 
+const MSG_STATUS = {
+  SUCCESS: 'SUCCESS',
+  PENDING: 'PENDING',
+  FAIL: 'FAIL',
+};
+
```
Comment (howieyoung) [inline src/Bubble.js]: So we are now recording status with these 3 categories for every message, right?

File: src/Bubble.js (hunk 2 above)
Comment (tungsheng) [reply src/Bubble.js]: Sure. Or 'Not Delivered'?

File: src/Bubble.js (hunk 1 above)
Comment (tungsheng) [reply src/Bubble.js]: OK

Comment (howieyoung) [reply src/Bubble.js]: @tungsheng fontSize could be 10

File: src/Bubble.js (hunk 3 above)
Comment (tungsheng) [reply src/Bubble.js]: Yes

File: src/Bubble.js (hunk 1 above)
Comment (tungsheng) [reply src/Bubble.js]: OK, i agree 9 is too small to read

File: src/Bubble.js (hunk 2 above)
Comment (howieyoung) [reply src/Bubble.js]: Actually I just thought that we should give it localization. Just like what we do for the image reminders.
 How can we achieve it?

Comment (tungsheng) [reply src/Bubble.js]: We can do localization setup in the gifted-chat package.

Comment (tungsheng) [reply src/Bubble.js]: Or do a 'Not Delivered' msg first. then localization in the future.

Comment (tungsheng) [reply src/Bubble.js]: https://www.google.com/search?q=not+delivered&source=lnms&tbm=isch&sa=X&ved=0ahUKEwjFzv7P87_hAhXIfLwKHUNtC68Q_AUIDigB&biw=1920&bih=1065

Comment (howieyoung) [reply src/Bubble.js]: or use props to transfer text from where we manage all text?

Comment (tungsheng) [reply src/Bubble.js]: that's not a good way to do this, but we can do this way first since it's the fastest implementation.

Comment (tungsheng) [reply src/Bubble.js]: I will push a commit later.

Comment (howieyoung) [reply src/Bubble.js]: Okay thanks, and yes it could be 'Not Delivered'

**Comment by tungsheng** [comment]
Empty string is falsy value, so this statement: '' || 'text', will always return 'text.

https://guide.freecodecamp.org/javascript/falsy-values/

**Comment by tungsheng** [comment]
If not empty string, it is a truthy value, so 'text1' || 'text2' will return 'text1'

**Pull request merged**